File: MesDonnes.m
//
//  MesDonnes.m
//  i-de
//
//  Created by DCF on 27/07/2014.
//  Copyright (c) 2014 Da Costa Faro Rémy. All rights reserved.
//

#import "MesDonnes.h"

@implementation MesDonnesNombreFace

- (void)encodeWithCoder:(NSCoder *)aCoder {
    [aCoder encodeInt:_nombreFace forKey:@"nombreFace"];
}

-(id)initWithCoder:(NSCoder *)aDecoder {
    _nombreFace = [aDecoder decodeIntForKey:@"nombreFace"];
    return self;
}

@end

@implementation MesDonnesLancerAuDemarrage

- (void)encodeWithCoder:(NSCoder *)aCoder {
    [aCoder encodeBool:_lancerAuDemarrage forKey:@"lancerAuDemarrage"];
}

-(id)initWithCoder:(NSCoder *)aDecoder {
    _lancerAuDemarrage = [aDecoder decodeBoolForKey:@"lancerAuDemarrage"];
    return self;
}

@end

@implementation MesDonnesAnimations

- (void)encodeWithCoder:(NSCoder *)aCoder {
    [aCoder encodeBool:_animations forKey:@"animations"];
}

-(id)initWithCoder:(NSCoder *)aDecoder {
    _animations = [aDecoder decodeBoolForKey:@"animations"];
    return self;
}

@end

#pragma mark - Secouer

@implementation MesDonnesSecouer

- (void)encodeWithCoder:(NSCoder *)aCoder {
    [aCoder encodeBool:_secouer forKey:@"secouer"];
}

-(id)initWithCoder:(NSCoder *)aDecoder {
    _secouer = [aDecoder decodeBoolForKey:@"secouer"];
    return self;
}

@end

@implementation MesDonnesSecouerRotation

- (void)encodeWithCoder:(NSCoder *)aCoder {
    [aCoder encodeBool:_secouerRotation forKey:@"secouerRotation"];
}

-(id)initWithCoder:(NSCoder *)aDecoder {
    _secouerRotation = [aDecoder decodeBoolForKey:@"secouerRotation"];
    return self;
}

@end

@implementation MesDonnesSecouerAnimations

- (void)encodeWithCoder:(NSCoder *)aCoder {
    [aCoder encodeBool:_secouerAnimations forKey:@"secouerAnimations"];
}

-(id)initWithCoder:(NSCoder *)aDecoder {
    _secouerAnimations = [aDecoder decodeBoolForKey:@"secouerAnimations"];
    return self;
}

@end

File: i-de/MesDonnes.swift
//
//  MesDonnes.swift
//  i-de
//
//  Created by remy on 10/11/2014.
//  Copyright (c) 2014 DCF. All rights reserved.
//

import UIKit

class MesDonnesAnimations : NSObject {
    var animations: Bool = true
    
    override init() {
        super.init()
    }
    
    func encodeWithCoder(aCoder: NSCoder) {
        aCoder.encodeBool(self.animations, forKey: "animations")
    }
    
    init(coder aDecoder: NSCoder) {
        super.init()
        self.animations = aDecoder.decodeBoolForKey("animations")
    }
}

// MARK: Secouer

class MesDonnesSecouer : NSObject {
    var secouer: Bool = true
    
    override init() {
        super.init()
    }
    
    func encodeWithCoder(aCoder: NSCoder) {
        aCoder.encodeBool(self.secouer, forKey: "secouer")
    }
    
    init(coder aDecoder: NSCoder) {
        super.init()
        self.secouer = aDecoder.decodeBoolForKey("secouer")
    }
}

class MesDonnesRotation : NSObject {
    var rotation: Bool = true
    
    override init() {
        super.init()
    }
    
    func encodeWithCoder(aCoder: NSCoder) {
        aCoder.encodeBool(self.rotation, forKey: "rotation")
    }
    
    init(coder aDecoder: NSCoder) {
        super.init()
        self.rotation = aDecoder.decodeBoolForKey("rotation")
    }
}

class MesDonnesSecouerAnimations : NSObject {
    var secouerAnimations: Bool = true
    
    override init() {
        super.init()
    }
    
    func encodeWithCoder(aCoder: NSCoder) {
        aCoder.encodeBool(self.secouerAnimations, forKey: "secouerAnimations")
    }
    
    init(coder aDecoder: NSCoder) {
        super.init()
        self.secouerAnimations = aDecoder.decodeBoolForKey("secouerAnimations")
    }
}